R/schema_Spreadsheet.R

```diff
@@ -0,0 +1,164 @@
+# https://developers.google.com/sheets/api/reference/rest/v4/spreadsheets#Spreadsheet
+new_Spreadsheet <- function(spreadsheetId = NULL,
+                            properties = NULL,
+                            sheets = NULL,
+                            namedRanges = NULL,
+                            spreadsheetUrl = NULL,
+                            developerMetadata = NULL) {
+  x <- list(
+    spreadsheetId = spreadsheetId,
+    properties = properties,
+    sheets = sheets,
+    namedRanges = namedRanges,
+    spreadsheetUrl = spreadsheetUrl,
+    developerMetadata = developerMetadata
+  )
+  structure(
+    validate_Spreadsheet(x),
+    class = "Spreadsheet"
+  )
+}
+
+validate_Spreadsheet <- function(x) {
+  maybe_string(x$spreadsheetId, "spreadsheetId")
+
+  if (!is.null(x$properties)) {
+    validate_SpreadsheetProperties(x$properties)
+  }
+
+  if (!is.null(x$sheets)) {
+    walk(x$sheets, validate_Sheet)
+  }
+
+  if (!is.null(x$namedRanges)) {
+    walk(x$namedRanges, validate_NamedRange)
+  }
+
+  maybe_string(x$spreadsheetUrl, "spreadsheetUrl")
+
+  # developerMetadata is an instance of DeveloperMetadata
+
+  x
+}
+
+Spreadsheet <- function(...) {
+  x <- new_Spreadsheet(...)
+  compact(x)
+}
+
+#  input: instance of Spreadsheet, in the Sheets API sense, as a named list
+# output: instance of sheets_Spreadsheet, which is how I want to hold this info
+sheets_Spreadsheet <- function(x = list()) {
```

Comment:
This is the pre-existing function I happened to already have for parsing an instance of `Spreadsheet` into the more natural R object.